File: app/src/main/java/govirtual360/firestoredemo/MainActivity.java
package govirtual360.firestoredemo;

import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.firebase.firestore.DocumentReference;
import com.google.firebase.firestore.DocumentSnapshot;
import com.google.firebase.firestore.EventListener;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.firestore.FirebaseFirestoreException;

import java.util.HashMap;
import java.util.Map;

public class MainActivity extends AppCompatActivity {
        EditText first,second;
        Button send,get;
    TextView textView;
    private final String TAG="data";
    private DocumentReference mdoc=FirebaseFirestore.getInstance().document("StoreDemo/Demo");

    @Override
    protected void onStart() {
        super.onStart();
        mdoc.addSnapshotListener(this, new EventListener<DocumentSnapshot>() {
            @Override
            public void onEvent(DocumentSnapshot documentSnapshot, FirebaseFirestoreException e) {
                if (documentSnapshot.exists()) {
                    textView.setText(documentSnapshot.getString("First")+"-"+documentSnapshot.getString("Second"));
                }
            }
        });
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        first=findViewById(R.id.editText);
        second=findViewById(R.id.editText2);
        send=findViewById(R.id.button);
        get=findViewById(R.id.button2);
        textView=findViewById(R.id.textView);

            send.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    Map<String, Object> user = new HashMap<>();
                    user.put("First", first.getText().toString());
                    user.put("Second", second.getText().toString());
                    mdoc.set(user).addOnSuccessListener(new OnSuccessListener<Void>() {
                        @Override
                        public void onSuccess(Void aVoid) {
                            Log.d(TAG,"Document saved");
                        }
                    }).addOnFailureListener(new OnFailureListener() {
                        @Override
                        public void onFailure(@NonNull Exception e) {
                            Log.d(TAG,"Document saved",e);
                        }
                    });
                }
            });

        get.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                mdoc.get().addOnSuccessListener(new OnSuccessListener<DocumentSnapshot>() {
                    @Override
                    public void onSuccess(DocumentSnapshot documentSnapshot) {
                        if (documentSnapshot.exists()) {
                            textView.setText(documentSnapshot.getString("First")+"-"+documentSnapshot.getString("Second"));
                        }

                    }
                });
            }
        });
    }


}

File: app/src/main/java/sample/firestoredemo/MainActivity.kt
package sample.firestoredemo

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast

import com.google.android.gms.tasks.OnFailureListener
import com.google.android.gms.tasks.OnSuccessListener
import com.google.firebase.firestore.DocumentReference
import com.google.firebase.firestore.DocumentSnapshot
import com.google.firebase.firestore.FirebaseFirestore

import java.util.HashMap

class MainActivity : AppCompatActivity() {
    internal lateinit var etFirstName: EditText
    internal lateinit var etMobileNo: EditText
    internal lateinit var btnSend: Button
    internal lateinit var btnGet: Button
    internal lateinit var tvUserDetails: TextView
    private val TAG = "data"
    private val mdoc = FirebaseFirestore.getInstance().document("student/studentData")

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        etFirstName = findViewById(R.id.et_name)
        etMobileNo = findViewById(R.id.et_mobileno)
        btnSend = findViewById(R.id.btn_send)
        btnGet = findViewById(R.id.btn_get)
        tvUserDetails = findViewById(R.id.tv_user_details)

        btnSend.setOnClickListener {
            val user = HashMap<String, Any>()
            user["name"] = etFirstName.text.toString()
            user["mobileNo"] = etMobileNo.text.toString()
            mdoc.set(user).addOnSuccessListener { Toast.makeText(this@MainActivity, "Student data saved successfully", Toast.LENGTH_SHORT).show() }.addOnFailureListener { Toast.makeText(this@MainActivity, "student error data came.", Toast.LENGTH_SHORT).show() }
        }

        btnGet.setOnClickListener {
            mdoc.get().addOnSuccessListener { documentSnapshot ->
                if (documentSnapshot.exists()) {
                    tvUserDetails.text = " Name - " + documentSnapshot.getString("name") + "\n Mobile No - " + documentSnapshot.getString("mobileNo")
                }
            }
        }
    }
}
